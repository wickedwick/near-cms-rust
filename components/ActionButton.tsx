import React from 'react'

const ActionButton = ({ children, onClick }: { children: React.ReactNode; onClick: () => void }) => {
  return (
    <button
      className="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
      onClick={onClick}
    >
      {children}
    </button>
  )
}

export default ActionButton
