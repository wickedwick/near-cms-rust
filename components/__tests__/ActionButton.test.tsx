import React from 'react'
import { configure, shallow } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import ActionButton from '../ActionButton'

configure({ adapter: new Adapter() });

const clickFunc = jest.fn()

describe('ActionButton', () => {
  it('should render', () => {
    const wrapper = shallow(<ActionButton onClick={clickFunc}>Text</ActionButton>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call onClick', () => {
    const wrapper = shallow(<ActionButton onClick={clickFunc}>Text</ActionButton>)
    wrapper.find('button').simulate('click')
    expect(clickFunc).toHaveBeenCalled()
  })
})
