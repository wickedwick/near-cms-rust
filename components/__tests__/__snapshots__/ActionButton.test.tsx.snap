// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton should render 1`] = `ShallowWrapper {}`;
